### PR TITLE
Transforms updates

### PR DESCRIPTION
**Pull request by joostvanzwieten**
The goal of this PR is to eliminate `PlainTransforms` in the `UnionTopology` and in the `boundary` and `interfaces` attributes of the `Topology` base class.

### CODE REVIEW

File: nutils/topology.py [hunk 1]
```diff
@@ -609,38 +608,63 @@ def boundary(self):
     '''
```
Comment (gertjanvanzwieten) [inline nutils/topology.py]: Should we use the opportunity to move this to ConnectedTopology?

File: nutils/topology.py [hunk 2]
```diff
@@ -609,38 +608,63 @@ def boundary(self):
...
+    if refs_touched:
+      references = elementseq.asreferences(references, self.ndims-1)
+    else:
+      references = self.references.edges[selection]
+    transforms = self.transforms.edges(self.references)[selection]
+    return Topology(references, transforms, transforms)
 
   @property
   @log.withcontext
   def interfaces(self):
```
Comment (gertjanvanzwieten) [inline nutils/topology.py]: idem boundary

File: tests/test_elementseq.py [hunk 3]
```diff
@@ -82,44 +81,68 @@ def test_mul_references(self):
...
-  check=[square*line]*2+[triangle*line]*2,
-  checkndims=3)
-common(
-  'ChildReferences',
-  seq=nutils.elementseq.ChildReferences(nutils.elementseq.PlainReferences([square, triangle], 2)),
-  check=[square]*4+[triangle]*4,
-  checkndims=2)
+class EmptyReferences(TestCase, Common):
+  @classmethod
+  def setUpClass(cls):
```
Comment (gertjanvanzwieten) [inline tests/test_elementseq.py]: why not just setUp instead of setUpClass? (applies to all following)

File: tests/test_transformseq.py [hunk 4]
```diff
@@ -153,115 +160,167 @@ def test_refined(self):
...
-common(
-  'StructuredTransforms:2D,refined',
-  seq=nutils.transformseq.StructuredTransforms(x2, [nutils.transformseq.DimAxis(0,2,False),nutils.transformseq.DimAxis(2,4,False)], 1),
-  check=[(x2,s01,c00),(x2,s01,c01),(x2,s01,c10),(x2,s01,c11)],
-  checkmissing=[(x2,s00,c00)],
-  checkrefs=nutils.elementseq.asreferences([square]*4, 2),
-  checkfromdims=2)
+class EmptyTransforms(TestCase, Common, Edges):
+  @classmethod
+  def setUpClass(cls):
```
Comment (gertjanvanzwieten) [inline tests/test_transformseq.py]: idem test_elementseq

File: tests/test_elementseq.py (hunk 3 above)
Comment (joostvanzwieten) [reply tests/test_elementseq.py]: done

File: tests/test_transformseq.py (hunk 4 above)
Comment (joostvanzwieten) [reply tests/test_transformseq.py]: done

File: nutils/topology.py (hunk 1 above)
Comment (joostvanzwieten) [reply nutils/topology.py]: That requires a bit more work than just moving the properties. The `ConnectedTopology` takes a precomputed connectivity. The `SimplexTopology` does not define a `boundary` or `interfaces` itself, but subclassing from `ConnectedTopology` requires computing the connectivity in `__init__`. We could add a, say, `ConnectedTopologyBase` with an abstract property `connectivity`, move the `boundary` and `interfaces` properties to this class and let `SimplexTopology` and `ConnectedTopology` derive from `ConnectedTopologyBase`.

Comment (gertjanvanzwieten) [reply nutils/topology.py]: I see, agreed. ConnectedTopologyBase seems to be the right approach but let's leave it for later.

**Pull request merged**